internal/provider/resource_vdb.go

```diff
@@ -881,6 +885,219 @@ func helper_provision_by_timestamp(ctx context.Context, d *schema.ResourceData,
 	return diags
 }
 
+func helper_provision_by_bookmark(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	client := meta.(*apiClient).client
+
+	provisionVDBFromBookmarkParameters := dctapi.NewProvisionVDBFromBookmarkParameters(d.Get("bookmark_id").(string))
+
+	// Setters for provisionVDBFromBookmarkParameters
+	if v, has_v := d.GetOk("source_data_id"); has_v {
+		provisionVDBFromBookmarkParameters.SetSourceDataId(v.(string))
+	}
```

Comment:
source_data_id is mutually exclusive with bookmark_id. That being said, maybe we can just let DCT complain if both are set?

Reply:
Yes, because now the VDB resource schema is combination of ProvisionVDBByTimestampParameters, ProvisionVDBBySnapshotParameters and ProvisionVDBFromBookmarkParameters. 

Some of the fields in these 3 request payloads will be mutually exclusive as you say. Some fields will be required in one of them but not in other two (i.e. source_data_id). So we should just let DCT complain if the invalid combination is set.